src/ramses_rf/device/hvac.py

```diff
@@ -375,7 +377,31 @@ def exhaust_temp(self) -> float | None:
 
     @property
     def fan_info(self) -> str | None:
-        return self._msg_value(Code._31DA, key=SZ_FAN_INFO)
+        """
+        Extract fan info description from _22F4, _31D9 or _31DA message payload
+
+        :return: string describing mode, speed
+        """
+        if (
+            Code._31D9 in self._msgs
+        ):  # Vasco D60 and ClimaRad minibox send mode/speed in _31D9
+            for k, v in self._msgs[Code._31D9].payload.items():
+                if (
+                    k == SZ_FAN_MODE and v != "FF"
+                ):  # Prevent ClimaRad Ventura constant "FF" to pass
+                    return str(v)
+            # no guard clause, just ignore
+        if Code._22F4 in self._msgs:  # ClimaRad Ventura sends mode/speed in _22F4
+            mode: str = ""
+            for k, v in self._msgs[Code._22F4].payload.items():
+                if k == SZ_FAN_MODE:
+                    mode = v
+                if k == SZ_FAN_RATE:
+                    mode = mode + ", " + v
+            return mode
+        return str(
+            self._msg_value(Code._31DA, key=SZ_FAN_INFO)
+        )  # a description to display in climate, e.g. "speed 2, medium", note localize in UI
```

Comment:
I'm sorry - this just looks wrong - you have to provide something that I can feel comfortable merging.

Not your fault - this is a symptom of the need for a Builder pattern.